tasks/partial_bundle.js

```diff
@@ -55,43 +93,56 @@ if(argv.length > 2) {
 function partialBundle(tasks, opts) {
     var name = opts.name;
     var index = opts.index;
+    var deleteIndex = opts.deleteIndex;
     var dist = opts.dist;
     var distMin = opts.distMin;
     var traceList = opts.traceList;
+    var transformList = opts.transformList;
+    var calendars = opts.calendars;
 
     tasks.push(function(done) {
         var partialIndex = mainIndex;
-        allTraces.forEach(function(trace) {
-            if(traceList.indexOf(trace) === -1) {
-                var WHITESPACE_BEFORE = '\\s*';
-                // remove require
-                var newCode = partialIndex.replace(
-                    new RegExp(
-                        WHITESPACE_BEFORE +
-                        'require\\(\'\\./' + trace + '\'\\),',
-                    'g'), ''
-                );
-
-                // test removal
-                if(newCode === partialIndex) throw 'Unable to find and drop require for trace: "' + trace + '"';
-
-                partialIndex = newCode;
+
+        var all = ['calendars'].concat(allTransforms).concat(allTraces);
+        var includes = (calendars ? ['calendars'] : []).concat(transformList).concat(traceList);
+        var excludes = all.filter(function(e) { return includes.indexOf(e) === -1; });
+
+        excludes.forEach(function(t) {
+            var WHITESPACE_BEFORE = '\\s*';
+            // remove require
+            var newCode = partialIndex.replace(
+                new RegExp(
+                    WHITESPACE_BEFORE +
+                    'require\\(\'\\./' + t + '\'\\)' +
+                    (t === 'calendars' ? '' : ','), // there is no comma after calendars require
```

Comment:
Feels a bit fragile - can we either just make the comma optional in the pattern, or add the comma in `lib/index` so all the lines are equivalent?